### DISCUSSION TITLE
PDA-P4 Support

### DISCUSSION THREAD

**Discussion by kamaleddin** [aqualinkd/AqualinkD#180] [Q&A]
Hello,

I haven' been successful in either running or compiling this project. Is my model, PDA-P4,  supported? It seems that the section regarding upgrading PDA chipsets has been removed from the wiki.

I was able to compile the slog separately and get this output. It seems that it does pick up my pump but it doesn't pick up my heater,  SWG and the light.

Great project and awesome work none the less.

Notice:  RS Serial: RS485 interface received 600 packets in 64 seconds (~9.38 Msg/Sec)
Notice:  RS Serial: Jandy Control Panel Model   :  PDA-P4 Only
Notice:  RS Serial: Jandy Control Panel Version :  PPD PDA. 5.0
Notice:  RS Serial: Jandy ID's found
Notice:  RS Serial: ID 0x78 is in use  <-- Jandy VSP ePump
Notice:  RS Serial: ID 0x60 is not used  <-- can use for Aqualinkd (PDA mode only)
Notice:  RS Serial: ID 0x61 is not used  <-- can use for Aqualinkd (PDA mode only)
Notice:  RS Serial: ID 0x62 is not used  <-- can use for Aqualinkd (PDA mode only)
Notice:  RS Serial: ID 0x63 is not used  <-- can use for Aqualinkd (PDA mode only)
Notice:  RS Serial: ID 0x60 is in use  <-- PDA Remote

**Comment by sfeakes** [comment]
serial logger will only pickup what's running on the RS485 buss, so light will never be seen (no lights are RS485), and SWG only if it's on and running at the same time slog is executing.  Heater would be seen if it's a RS485 heater (many are not wired up in that mode even if they support it).
So your output looks good.  You should be able to use that information to configure aqualinkd.conf and start from there.

The id 0x60 is listed twice, so the PDA remote was probably in sleep mode then woke while slog was being executed.

**Reply by kamaleddin** [reply]
Thanks a lot for the reply. I keep getting the following error when I try to compile the project. not sure what I am doing wrong
kamaleddin@poolpi:~/software/AqualinkD $ make 
OS: 11 (bullseye) 
GLIBC: ldd (Debian GLIBC 2.31-13+rpt2+rpi1+deb11u2) 2.31 
gcc -Wall -O3  -D AQ_PDA -D MG_DISABLE_MD5 -D MG_DISABLE_HTTP_DIGEST_AUTH -D MG_DISABLE_MD5 -D MG_DISABLE_JSON_RPC  -c aqualinkd.c  -o aqualinkd.o
aqualinkd.c: In function ‘main_loop’:
aqualinkd.c:1493:19: error: ‘struct aqconfig’ has no member named ‘extended_device_id’
 1493 |     if (_aqconfig_.extended_device_id != 0x00)
      |                   ^
aqualinkd.c:1495:136: error: ‘struct aqconfig’ has no member named ‘extended_device_id’
 1495 |       LOG(AQUA_LOG,LOG_ERR, "Aqualink daemon can't use extended_device_id in PDA mode, ignoring value '0x%02hhx' from cfg\n",_aqconfig_.extended_device_id);
      |                                                                                                                                        ^
aqualinkd.c:1496:17: error: ‘struct aqconfig’ has no member named ‘extended_device_id’
 1496 |       _aqconfig_.extended_device_id = 0x00;
      |                 ^
aqualinkd.c:1497:17: error: ‘struct aqconfig’ has no member named ‘extended_device_id_programming’
 1497 |       _aqconfig_.extended_device_id_programming = false;
      |                 ^
make: *** [Makefile:175: aqualinkd.o] Error 1

**Comment by sfeakes** [comment]
Since you are using a PI, the binary’s in the repo should work.

but the error looks like you have an old or missing header,  have you modified anything from the repo / got a clean copy?

**Comment by ballle98** [comment]
Somewhat a duplicate of  #155.  I can see that you are trying to compile under OS: 11 (bullseye) and that is known not to compile.  I have a pull request to fix this #156 as well as another request to fix new warnings in bullseye #174 

You can try my fork with these pulls merged https://github.com/ballle98/AqualinkD/wiki#install

It also look like you are not using the default "make clean; make"

You have different -D defines
`gcc -Wall -O3 -D AQ_PDA -D MG_DISABLE_MD5 -D MG_DISABLE_HTTP_DIGEST_AUTH -D MG_DISABLE_MD5 -D MG_DISABLE_JSON_RPC -c aqualinkd.c -o aqualinkd.o`

Than what I see when I build:
`gcc -Wall -O3  -D AQ_PDA -D AQ_ONETOUCH -D AQ_IAQTOUCH -D AQ_RS16 -D MG_DISABLE_MD5 -D MG_DISABLE_HTTP_DIGEST_AUTH -D MG_DISABLE_MD5 -D MG_DISABLE_JSON_RPC  -c aqualinkd.c  -o aqualinkd.o
`

**Reply by kamaleddin** [reply]
thank you, sir. I already used your fork and it worked incredibly well. Also, I was the one who opened the SWG issue for your fork:D
